### DISCUSSION TITLE
Guidance on continuous test (loop) and collecting information along the way

### DISCUSSION THREAD

**Discussion by iceman8420** [seleniumbase/SeleniumBase#2420] [Q&A] [answered]
First of all, heartfelt thank you to Michael for this amazing automation framework, additionally the community behind it.

I'm humbly looking for some guidance or direction - I'm very familiar with scripting in powershell (please don't judge me for that :) ), but recently picked up python and ported my code over in order to take advantage of seleniumbase and pytest, specifically because of UC mode capabilities required to successfully test what I need.

Trying to be brief and general --- I have a use case where I want to: 

- check a web page for content on a regular interval
- when a new object appears on that web page for the first time, I want to collect details on that object and take action.  For the sake of argument, let's say I want to send an email about that new content, but only the first time it appears
- on subsequent page checks, I don't want another action for that particular new object until the next day - it should be ignored until the end of the day

I have this code working perfectly with regular python.  With an an endless loop, I check this page, do something when new objects show up, and then 'clear' the alerts at the end of the day.  

I also successfully ported the tests over to the pytest methods as I want to take advantage of those extra features like hover and click.

My ask / Where I need some guidance -- I have successful pytest code doing tests in one iteration, however, I'm not sure how to 'collect' information from the test and use that information on subsequent loops in order to only take action once per object that appears.   I tried some examples of paramterization and pytest but can't seem to find the right combo.

Any pointers or help on what to research is appreciated.

**Comment by mdmintz** [comment, marked as answer]
Hello, there are some methods for saving and reading data to/from files. Use these in combination with flow-of-control statements (Eg. `if`):

```python
self.save_data_as(data, file_name, destination_folder=None)

self.append_data_to_file(data, file_name, destination_folder=None)

self.get_file_data(file_name, folder=None)
```

**Reply by iceman8420** [reply]
I can make that work, dumping out to a file system will do just right. Perfect. THANK YOU for the pointer!

edit: not sure if it is by design -- but save_data_as has slightly different behavior than append_data_to_file, on an empty file.

Example, if I have a list - somelist 
`self.append_data_to_file(somelist, 'testout_append.py', destination_folder=None)
                self.save_data_as(somelist, 'testout_saveas.py', destination_folder=None)`

Produces this in testout_append.py:
`['ListItem1']`

And this in testout_saveas.py:
`ListItem1`
